probe.go

```diff
@@ -383,15 +385,31 @@ func (p *Probe) internalInit(manager *Manager) error {
 		}
 	}
 
-	// Find function name match if required
-	if p.MatchFuncName != "" && len(p.HookFuncName) == 0 {
-		// if this is a kprobe or a kretprobe, look for the symbol now
-		if p.GetKprobeType() != UnknownProbeType {
-			var err error
-			p.HookFuncName, err = FindFilterFunction(p.MatchFuncName)
-			if err != nil {
-				p.lastError = err
-				return err
+	if p.programSpec.Type == ebpf.Kprobe {
+		progType, _, _ := strings.Cut(p.programSpec.SectionName, "/")
+		switch progType {
+		case "kprobe":
+			p.kprobeType = kprobe
+		case "kretprobe":
+			p.kprobeType = kprobe
+			p.isReturnProbe = true
```

Comment:
Why not just use a different type and remove `isReturnProbe`?

Reply:
because the type (kprobe/uprobe) determines which file to use, while the entry/return determines the the format of the line.